pkg/autoscaler/autoscaler.go

```diff
@@ -141,6 +191,50 @@ func (agg *totalAggregation) observedConcurrencyPerPod(now time.Time) float64 {
 	return accumulatedConcurrency / observedPods
 }
 
+// The estimated concurrency per pod (average of concurrencies of all
+// non-lameduck samples)
+// Ignores activator sent metrics if its not the only pod reporting stats
+func (agg *totalAggregation) estimatedConcurrencyPerPod(now time.Time, logger *zap.SugaredLogger) float64 {
```

Comment:
I don't understand why you need an `estimated*` version of all the aggregation functions.  I would expect that you would just put in the stats you have and get the average, same as before.  Is there something subtle I'm missing?

Reply:
What `estimated*` version of all the aggregation functions do are what you expected: put in the stats I have and get the average. Do you think the existing aggregation functions also suit the data if we are doing sampling?

Reply:
The existing aggregation functions are based on all data from all pods. We aggregate on pod level first then get the average. When we are doing sampling, IMO we can just get the average regardless which pod the sample is from(except activator). No need to aggregate on pod level first. WDYT?

Reply:
> Do you think the existing aggregation functions also suit the data if we are doing sampling?

Yes.  We just need to provide the current number of "in-service" pods when calculating the average.  (See my comment above)

Reply:
> No need to aggregate on pod level first.

This was added to allow new pods to have equal "weight" to old pods.  This prevents a runaway situation where adding new pods doesn't seem to affect the average much so we add more pods.